### DISCUSSION TITLE
How to create a split violin chart?

### DISCUSSION THREAD

**Discussion by dheeraj-murali** [airbnb/visx#1288] [Q&A]
Hi all, I'm new to visx. I've been trying to figure out a way to to plot a split violin chart, something that looks like this

![Screenshot 2021-07-22 at 20 05 09](https://user-images.githubusercontent.com/17213265/126657274-10d9e618-0e05-4ec8-a0d5-b01d8779342c.png)

using `@visx/stats` `<ViolinPlot />` and I'm not able to figure it out. It could be that I might be looking at the wrong place. 
Any insight into this or ideas on how to plot a chart like this?

Thanks for all the work everyone has put into it. Visx is great to work with!

**Comment by williaster** [comment]
Hey @dheeraj-murali 👋  thanks for checking out `visx`, happy to hear you dig it so far 😎 

To achieve something like this, I think you'd probably need to render **two violin plots on top of each other**, and then use [`clip-path`s](https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Clipping_and_masking) to only show half of them. 

I forked the [stats-plot demo](https://airbnb.io/visx/statsplot) and used `@visx/clip-path` to achieve this. Hopefully you could tweak that and land on something that works for you! It's a little extra complex because of the patterns, if you removed them it shouldn't be too complex; it also doesn't have an axis or grid but you can use `@visx/axis` + `@visx/grid` for that. 

Here's the [link](https://codesandbox.io/s/strange-rgb-fdfy5?file=/Example.tsx).

<img src="https://user-images.githubusercontent.com/4496521/126710499-80684249-e57c-4606-9bc1-36dc9e2e8d1d.png" width="500" />

**Reply by dheeraj-murali** [reply]
Thank you @williaster for the quick response, this one works wonderfully. 

I do have one more question, how do I add the line that you see on the violin chart that represents the Quartile values.
I tried using `@visx/marker` but not able to figure out the coordinates to plot the line.
